common/installshieldv3_archive.cpp

```diff
@@ -107,7 +107,8 @@ void InstallShieldV3::close() {
 	_map.clear();
 }
 
-bool InstallShieldV3::hasFile(const Common::String &name) const {
+bool InstallShieldV3::hasFile(const Common::Path &path) const {
+	Common::String name = path.toString();
```

Comment:
All of this .toString()-inging, can't that be done inline in the various callsites? (Or where it makes sense, possibly make the called function take a Path, if that's the only thing it needs to deal with anyhow)

Reply:
I tried doing that initially, but I found that in a lot of cases, it resulted in code much uglier (and slightly less efficient) than stringifying the path once at the beginning of the function.

As for the necessity of stringifying the paths in the first place, there are a ton of `Archive` subclasses which were designed to use strings as input. Ideally, they should all be changed to fully work with paths, but doing that would require modifying *a lot* of code I'm unfamiliar with, which would eat up my remaining GSoC time and possibly introduce bugs. Stringifying the path and leaving the rest of the code as-is guarantees that the code's behavior will be the same as before, so I don't think it's worth the effort of making further changes to most of these `Archive` subclasses right now.